packages/block-library/src/icon/components/icon-dropzone/index.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { DropZone } from '@wordpress/components';
+import { isBlobURL } from '@wordpress/blob';
+
+/**
+ * Internal dependencies
+ */
+import {
+	parseUploadedMediaAndSetIcon,
+	parseDroppedMediaAndSetIcon,
+	displayMessages,
+} from './../../utils';
+
+export default function IconDropZone( props ) {
+	const { attributes, setAttributes, mediaUpload, isSVGUploadAllowed } =
+		props;
+
+	function onDropFiles( filesList ) {
+		if ( isSVGUploadAllowed ) {
+			mediaUpload( {
+				allowedTypes: [ 'image/svg+xml' ],
+				filesList,
+				onFileChange( [ image ] ) {
+					if ( isBlobURL( image?.url ) ) {
+						return;
+					}
+					parseUploadedMediaAndSetIcon(
+						image,
+						attributes,
+						setAttributes
+					);
+				},
+				onError() {
+					displayMessages( 'fileTypeUploadError' );
+				},
```

Comment:
```suggestion
				onError( message ) {
					createErrorNotice( message, { type: 'snackbar' } );
				},
```

Can't we just use the error message from the mediaUpload component itself?